setup.py

```diff
@@ -23,7 +23,7 @@
         "Topic :: Scientific/Engineering",
     ],
     packages=find_packages(),
-    python_requires=">=3.8",
+    python_requires=">=3.8", #! should we update this in the environment.yml file?
```

Comment:
There are a lot of TODOs and "questions to self" being checked into main that would probably be better kept as notes on a private branch, or, better yet, issues and discussion questions in github.

Reply:
Agreed, we should probably resolve and remove the open comments before we merge this into main.